.github/workflows/performance.yml

```diff
@@ -0,0 +1,160 @@
+
+      - name: Docker debug information
+        run: |
+          docker -v
+          docker-compose -v
+          docker-compose run --rm mysql mysql --version
+          docker-compose run --rm php php --version
+          docker-compose run --rm php php -m
+          docker-compose run --rm php php -i
+          docker-compose run --rm php locale -a
+
+      - name: Install WordPress
+        run: npm run env:install
+
+      - name: Install WordPress Importer plugin and import mock data
+        run: |
+          npm run env:cli -- plugin install wordpress-importer --activate
+          curl -O https://raw.githubusercontent.com/WPTT/theme-test-data/b9752e0533a5acbb876951a8cbb5bcc69a56474c/themeunittestdata.wordpress.xml
+          npm run env:cli -- import themeunittestdata.wordpress.xml --authors=create
+          rm themeunittestdata.wordpress.xml
+
+      - name: Update permalink structure
+        run: |
+          npm run env:cli -- rewrite structure '/%year%/%monthnum%/%postname%/'
+
+      - name: Install MU plugin
+        run: |
+          mkdir ./build/wp-content/mu-plugins
+          cp ./tests/performance/mu-plugins/server-timing.php ./build/wp-content/mu-plugins/server-timing.php
+
+      - name: Run Performance tests (current commit)
+        run: npm run test:performance
+
+      - name: Print Performance tests results
+        run: "node ./tests/performance/results.js"
+
+      - name: Update the environment to 6.1.1
+        run: |
+          npm run env:cli -- core update --version=6.1.1 --force --path=/var/www/build
+          npm run env:cli -- core version --path=/var/www/build
+
+      - name: Run Performance tests (6.1.1)
+        run: npm run test:performance -- --prefix=base
+
+      - name: Print Performance tests results (6.1.1)
+        run: "node ./tests/performance/results.js --prefix=base"
+
+      - name: Set commit details
+        uses: actions/github-script@98814c53be79b1d30f795b907e553d8679345975 # v6.4.0
+        id: commit-timestamp
+        with:
+            github-token: ${{secrets.GITHUB_TOKEN}}
+            script: |
+                const commit_details = await github.rest.git.getCommit({owner: context.repo.owner, repo: context.repo.repo, commit_sha: context.sha});
+                return parseInt((new Date( commit_details.data.author.date ).getTime() / 1000).toFixed(0))
+
+      - name: Publish performance results
+        # Only publish results on pushes to trunk.
+        if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/trunk' }}
+        env:
+            COMMITTED_AT: ${{ steps.commit-timestamp.outputs.result }}
+            CODEVITALS_PROJECT_TOKEN: ${{ secrets.CODEVITALS_PROJECT_TOKEN }}
+            HOST_NAME: "codevitals.run"
+        # The base hash hard coded here is the commit to the WP 6.1.1 tag.
+        run: node ./tests/performance/log-results.js $CODEVITALS_PROJECT_TOKEN trunk $GITHUB_SHA 031cedc06d612c4a74f7b6d3e77d8145f2954a25 $COMMITTED_AT $HOST_NAME
```

Comment:
If we make the workflow more dynamic in the other areas, would we be able to pass a branch or tag name instead of a SHA value here? If not, then this SHA would also need to be fetched through a REST API call, or something similar.

Reply:
This value is used for logging purposes, but if we do want to make the baseline version variable, we could use the `gitTag` method on the API Client to get this value based on a tag reference.

See: https://octokit.github.io/rest.js/v19#git-get-tag

Reply:
Handled in [04c0e39](https://github.com/WordPress/wordpress-develop/pull/4139/commits/04c0e395fd1c6e72a29fd3809d9c0b9b6943b3f0)